scripts/Muon/GUI/ElementalAnalysis/elemental_analysis.py

```diff
@@ -388,22 +396,26 @@ def select_data_file(self):
             self.ptable.set_peak_datafile(filename)
 
         try:
-            self._generate_element_widgets()
+            any_data_loaded = self._generate_element_widgets()
+            if not any_data_loaded:
+                self._reset_data_file_warning_and_action()
         except ValueError:
-            message_box.warning(
-                'The file does not contain correctly formatted data, resetting to default data file.'
-                'See "https://docs.mantidproject.org/nightly/interfaces/'
-                'Muon%20Elemental%20Analysis.html" for more information.')
-            self.ptable.set_peak_datafile(None)
-            self._generate_element_widgets()
-
+            self._reset_data_file_warning_and_action()
         for element in old_lines:
             if element in self.element_widgets.keys():
                 self.ptable.select_element(element)
             else:
                 self._remove_element_lines(element)
         self._update_checked_data()
 
+    def _reset_data_file_warning_and_action(self):
+        message_box.warning(
+            'The file does not contain correctly formatted data, resetting to default data file.'
+            'See "https://docs.mantidproject.org/nightly/interfaces/'
+            'Muon%20Elemental%20Analysis.html" for more information.')
```

Comment:
The URL is incorrect. It should be https://docs.mantidproject.org/nightly/interfaces/muon/Muon%20Elemental%20Analysis.html (there is a folder for muon interfaces within the interfaces folder)

Reply:
```suggestion
            'See "https://docs.mantidproject.org/nightly/interfaces/muon/'
            'Muon%20Elemental%20Analysis.html" for more information.')
```